### PR TITLE
saul-41

### PR DESCRIPTION
**Pull request by NahuelBerg**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/tests/latest/test_ams_content_key_policy_scenarios.py [hunk 1]
```diff
@@ -71,3 +70,81 @@ def test_content_key_policy_show_basic(self, storage_account_for_show):
             self.check('options[0].configuration.odatatype', '{configurationODataType}'),
             self.check('options[0].restriction.odatatype', '{restrictionODataType}')
         ])
+
+    @ResourceGroupPreparer()
+    @StorageAccountPreparer(parameter_name='storage_account_for_create')
+    def test_content_key_policy_delete(self, storage_account_for_create):
```
Comment (Renkon) [inline src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/tests/latest/test_ams_content_key_policy_scenarios.py]: event is for delete yet storage account is for create.. silly thing but we need to make it consistent

File: src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/tests/latest/test_ams_content_key_policy_scenarios.py [hunk 2]
```diff
@@ -71,3 +70,81 @@ def test_content_key_policy_show_basic(self, storage_account_for_show):
...
+            'description': 'ExampleDescription',
+            'policyOptionName': policy_option_name,
+            'configurationODataType': '#Microsoft.Media.ContentKeyPolicyClearKeyConfiguration',
+            'restrictionODataType': '#Microsoft.Media.ContentKeyPolicyOpenRestriction'
+        })
+
+        self.cmd('az ams account create -n {amsname} -g {rg} --storage-account {storageAccount} -l {location}')
+
+        self.cmd('az ams content-key-policy create -a {amsname} -n {contentKeyPolicyName} -g {rg} --description {description} --clear-key-configuration --open-restriction --policy-option-name {policyOptionName}', checks=[
+            self.check('name', '{contentKeyPolicyName}')
```
Comment (Renkon) [inline src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/tests/latest/test_ams_content_key_policy_scenarios.py]: unneeded test

File: src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/tests/latest/test_ams_content_key_policy_scenarios.py [hunk 3]
```diff
@@ -71,3 +70,81 @@ def test_content_key_policy_show_basic(self, storage_account_for_show):
...
+    @StorageAccountPreparer(parameter_name='storage_account_for_create')
+    def test_content_key_policy_list(self, storage_account_for_create):
+        amsname = self.create_random_name(prefix='ams', length=12)
+        policy_name = self.create_random_name(prefix='pn', length=12)
+        policy_name2 = self.create_random_name(prefix='pn', length=10)
+        policy_option_name = self.create_random_name(prefix='pon', length=12)
+
+        self.kwargs.update({
+            'amsname': amsname,
+            'storageAccount': storage_account_for_create,
```
Comment (Renkon) [inline src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/tests/latest/test_ams_content_key_policy_scenarios.py]: idem comment 1

File: src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/tests/latest/test_ams_content_key_policy_scenarios.py [hunk 4]
```diff
@@ -71,3 +70,81 @@ def test_content_key_policy_show_basic(self, storage_account_for_show):
...
+            'description': 'ExampleDescription',
+            'policyOptionName': policy_option_name,
+            'configurationODataType': '#Microsoft.Media.ContentKeyPolicyClearKeyConfiguration',
+            'restrictionODataType': '#Microsoft.Media.ContentKeyPolicyOpenRestriction'
+        })
+
+        self.cmd('az ams account create -n {amsname} -g {rg} --storage-account {storageAccount} -l {location}')
+
+        self.cmd('az ams content-key-policy create -a {amsname} -n {contentKeyPolicyName} -g {rg} --description {description} --clear-key-configuration --open-restriction --policy-option-name {policyOptionName}', checks=[
+            self.check('name', '{contentKeyPolicyName}')
```
Comment (Renkon) [inline src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/tests/latest/test_ams_content_key_policy_scenarios.py]: idem comment 2

**Review by damatojuan — APPROVED** [review]
I would merge list & delete tests into one.
It's not necessary to add checkers to commands that are not being tested (e.g. create)

File: src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/tests/latest/test_ams_content_key_policy_scenarios.py [hunk 5]
```diff
@@ -71,3 +70,81 @@ def test_content_key_policy_show_basic(self, storage_account_for_show):
...
+        ])
+
+        self.cmd('az ams content-key-policy list -a {amsname} -g {rg}', checks=[
+            self.check('length(@)', 1)
+        ])
+
+        self.cmd('az ams content-key-policy create -a {amsname} -n {contentKeyPolicyName2} -g {rg} --description {description} --clear-key-configuration --open-restriction --policy-option-name {policyOptionName}', checks=[
+            self.check('name', '{contentKeyPolicyName2}')
```
Comment (Renkon) [inline src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/tests/latest/test_ams_content_key_policy_scenarios.py]: idem comment 2

**Review by Renkon — CHANGES_REQUESTED** [review]

**Pull request merged**